src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            public unsafe sealed record SymUnmanagedMethodRcw(IntPtr Inst) : ISymUnmanagedMethod
+            {
+                private bool _disposed = false;
+
+                public int GetToken(out int methodToken)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int*, int>)(*(*(void***)Inst + 3));
+                    fixed (int* methodTokenPtr = &methodToken)
+                    {
+                        return func(Inst, methodTokenPtr);
+                    }
+                }
+
+                public int GetSequencePointCount(out int count)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int*, int>)(*(*(void***)Inst + 4));
+                    fixed (int* countPtr = &count)
+                    {
+                        return func(Inst, countPtr);
+                    }
+                }
+
+                public int GetRootScope(out ISymUnmanagedScope? scope)
+                {
+                    var func = (delegate* unmanaged<IntPtr, IntPtr*, int>)(*(*(void***)Inst + 5));
+                    IntPtr scopePtr;
+                    int hr = func(Inst, &scopePtr);
+                    scope = hr == 0
+                        ? (SymUnmanagedScopeWrapperCache.SymUnmanagedScopeRcw)SymUnmanagedScopeWrapperCache.Instance.GetOrCreateObjectForComInstance(scopePtr, CreateObjectFlags.UniqueInstance)
+                        : null;
+                    return hr;
+                }
+
+                public int GetScopeFromOffset(int offset, out ISymUnmanagedScope? scope)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, IntPtr*, int>)(*(*(void***)Inst + 6));
+                    IntPtr scopePtr;
+                    int hr = func(Inst, offset, &scopePtr);
+                    scope = hr == 0
+                        ? (SymUnmanagedScopeWrapperCache.SymUnmanagedScopeRcw)SymUnmanagedScopeWrapperCache.Instance.GetOrCreateObjectForComInstance(scopePtr, CreateObjectFlags.UniqueInstance)
+                        : null;
+                    return hr;
+                }
+
+                public int GetOffset(ISymUnmanagedDocument document, int line, int column, out int offset)
+                {
+                    var func = (delegate* unmanaged<IntPtr, ISymUnmanagedDocument, int, int, int*, int>)(*(*(void***)Inst + 7));
+                    fixed (int* offsetPtr = &offset)
+                    {
+                        return func(Inst, document, line, column, offsetPtr);
+                    }
+                }
+
+                public int GetRanges(ISymUnmanagedDocument document, int line, int column, int bufferLength, out int count, int[] ranges)
+                {
+                    var func = (delegate* unmanaged<IntPtr, ISymUnmanagedDocument, int, int, int, int*, int*, int>)(*(*(void***)Inst + 8));
```

Comment:
Oops I missed this one. Will change to use `GCHandle` for `ISymUnmanagedDocument` marshalling.